examples/compute/src/shared/cv/bir_cv_memScript.sml

```diff
@@ -0,0 +1,209 @@
+End
+
+Definition cv_n2v_def:
+  cv_n2v n = boolify [] (cv_n2l n)
+End
+
+
+(* Load a bitstring at the given address from a mmap and pad it *)
+Definition bir_cv_load_bitstring_from_mmap_def:
+  bir_cv_load_bitstring_from_mmap vty mmap (a:num) =
+    fixwidth (size_of_bir_immtype vty) (cv_n2v (bir_cv_load_mmap mmap a))
+End
+  
+(* Load n splits of size vty from mmap starting at addr *) 
+Definition bir_cv_load_splits_from_mmap_def:
+  bir_cv_load_splits_from_mmap aty vty mmap addr n =
+    (MAP (\k. bir_cv_load_bitstring_from_mmap vty mmap (bir_mem_addr aty (addr + k))) (COUNT_LIST n)) 
+End
+
+
+(* Computes an already unpacked load expression *)
+Definition bir_cv_compute_load_from_mem_def:
+  bir_cv_compute_load_from_mem
+  (vty : bir_immtype_t) (rty : bir_immtype_t) (aty : bir_immtype_t) (mmap_alist : (num # num) list) (en: bir_endian_t) (addr:num) =
+
+   case (bir_number_of_mem_splits vty rty aty) of
+    | NONE => NONE
+    | SOME (n:num) => (
+        let vs = bir_cv_load_splits_from_mmap aty vty mmap_alist addr n in
+        let vs' = (case en of BEnd_LittleEndian => SOME (REVERSE vs)
+                          |  BEnd_BigEndian => SOME vs
+                          |  BEnd_NoEndian => if (n = 1) then SOME vs else NONE) in
+        case vs' of NONE => NONE
+                 |  SOME vs'' => SOME (bir_mem_concat vs'' rty)
+   )
+End
+
+Definition bir_cv_compute_load_def:
+  (bir_cv_compute_load (SOME (BCVVal_Mem aty vty mmap_alist)) (SOME (BCVVal_Imm addr)) en rty = 
+    cv_val_from_imm_option (bir_cv_compute_load_from_mem vty rty aty mmap_alist en (b2n addr))) /\
+  (bir_cv_compute_load _ _ _ _ = NONE)
+End
+
+
+(* ----------------------------------------- *)
+(* ----------------- STORE ----------------- *)
+(* ----------------------------------------- *)
+
+
+(* Add all the bitstrings in the mmap at address a *)
+Definition bir_cv_update_mmap_def:
+  (bir_cv_update_mmap aty mmap_alist a [] = mmap_alist) /\
+  (bir_cv_update_mmap aty mmap_alist a (v::vs) =
+    bir_cv_update_mmap aty (((bir_mem_addr aty a), v2n v)::mmap_alist) (SUC a) vs)
+End
+
+
+(* Compute an already unpacked store expression *)
+Definition bir_cv_compute_store_in_mem_def:
+  bir_cv_compute_store_in_mem
+  (vty : bir_immtype_t) (aty : bir_immtype_t) (result : bir_imm_t) (mmap_alist : (num # num) list) (en: bir_endian_t) (addr:num) =
+
+   let rty = type_of_bir_imm result in
+   case (bir_number_of_mem_splits vty rty aty) of
+    | NONE => NONE
+    | SOME (n:num) => (
+      case  (bitstring_split (size_of_bir_immtype vty) (b2v result)) of
+        | NONE => NONE
+        | SOME vs =>
+        let vs' = (case en of BEnd_LittleEndian => SOME (REVERSE vs)
+                          |  BEnd_BigEndian => SOME vs
+                          |  BEnd_NoEndian => if (n = 1) then SOME vs else NONE) in
+
+        case vs' of NONE => NONE
+                 |  SOME vs'' => SOME (BCVVal_Mem aty vty (bir_cv_update_mmap aty mmap_alist addr vs''))
+   )
```

Comment:
@jules-timmerman note that you can do `case ... of` on multiple things at once to simplify the code, e.g., 
```sml
    case (FLOOKUP m1 k, FLOOKUP m2 k) of
    | (SOME v1, SOME v2) => SOME $ f v1 v2
    | (f1, f2) => OPTION_CHOICE f1 f2
```